src/smefit/optimize/ultranest.py

```diff
@@ -2,15 +2,22 @@
 """Fitting the Wilson coefficients with |NS|"""
 import time
 
+from functools import partial
 import ultranest
 from rich.style import Style
 from rich.table import Table
 from ultranest import stepsampler
 
+import jax
+import jax.numpy as jnp
+
 from .. import log
 from ..coefficients import CoefficientManager
 from ..loader import load_datasets
 from . import Optimizer
+from .. import chi2
+
+jax.config.update("jax_enable_x64", True)
```

Comment:
For me it actually didn't give any issues and I even got an additional speed up: 15.770 minutes on 32 bit versus 23.960 minutes on 64 bit. The results haven't changed:
 
![coefficient_histo](https://github.com/LHCfitNikhef/smefit_release/assets/54140851/1c9b8b61-7dba-4edc-95e7-bdd794a00869)

Reply:
Did you try on a number of operators lower than 10? I think the stepsampler works but the ellipsoid one does not?

Yeah, I don't know about mpi...

Reply:
Ah when I try with for instance three operators I get an error: 

`  File "ultranest/mlfriends.pyx", line 1451, in ultranest.mlfriends.WrappingEllipsoid.inside
  File "ultranest/mlfriends.pyx", line 690, in ultranest.mlfriends._inside_ellipsoid
ValueError: Buffer dtype mismatch, expected 'float_t' but got 'float'`

So you're right, the ellipsoid sampler complains about an incorrect type.

Reply:
Yes, I opened an issue in the ultranest repo and the problem is that some stuff is precompiled and it assumes float64, so there seems to be no way out in this sense. What could be cool maybe is to have a switch so that we can control it and use float32?

Reply:
Good idea, let's make this line conditional on the number of fitting parameters. Is 10 the threshold?

Reply:
Done! Double precision is now an option in the run card, and is always assumed below 11 dof